File: 백준/Silver/10845. 큐/큐.py
from sys import stdin
from collections import deque

n = int(stdin.readline())
instructions = [stdin.readline().split() for _ in range(n)]
queue = deque()

for instruction in instructions:
    if instruction[0] == 'push':
        queue.append(instruction[1])
    elif instruction[0] == 'pop':
        print(queue.popleft() if queue else -1)
    elif instruction[0] == 'size':
        print(len(queue))
    elif instruction[0] == 'empty':
        print(0 if queue else 1)
    elif instruction[0] == 'front':
        print(queue[0] if queue else -1)
    elif instruction[0] == 'back':
        print(queue[-1] if queue else -1)

File: 백준/Silver/10845. 큐/큐.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        String[] instructions = new String[n];
        for (int i = 0; i < n; i++) {
            instructions[i] = br.readLine();
        }

        Deque<String> queue = new ArrayDeque<>();
        for (String instruction : instructions) {
            String[] splitInstruction = instruction.split(" ");
            switch (splitInstruction[0]) {
                case "push":
                    queue.offer(splitInstruction[1]);
                    break;
                case "pop":
                    System.out.println(queue.isEmpty() ? -1 : queue.pollFirst());
                    break;
                case "size":
                    System.out.println(queue.size());
                    break;
                case "empty":
                    System.out.println(queue.isEmpty() ? 1 : 0);
                    break;
                case "front":
                    System.out.println(queue.isEmpty() ? -1 : queue.peekFirst());
                    break;
                case "back":
                    System.out.println(queue.isEmpty() ? -1 : queue.peekLast());
                    break;
            }
        }
    }
}